File: backend/app/schemas/analytics.py
from pydantic import BaseModel, Field
from typing import Optional

class AnalyticsSummary(BaseModel):
    total_complaints: int = Field(default=0, alias="totalComplaints")
    open_complaints: int = Field(default=0, alias="openComplaints")
    in_progress_complaints: int = Field(default=0, alias="inProgressComplaints")
    resolved_complaints: int = Field(default=0, alias="resolvedComplaints")
    deleted_complaints: int = Field(default=0, alias="deletedComplaints")

    class Config:
        populate_by_name = True

class CategoryCount(BaseModel):
    category: str
    count: int

class PriorityCount(BaseModel):
    priority: str
    count: int

class TrendDataPoint(BaseModel):
    date: str
    count: int

class HeatmapPoint(BaseModel):
    latitude: float
    longitude: float
    category: str
    priority: str
    status: str

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  totalComplaints: number;
  openComplaints: number;
  inProgressComplaints: number;
  resolvedComplaints: number;
  deletedComplaints: number;
}

export interface CategoryCount {
  category: string;
  count: number;
}

export interface PriorityCount {
  priority: string;
  count: number;
}

export interface TrendDataPoint {
  date: string;
  count: number;
}

export interface HeatmapPoint {
  latitude: number;
  longitude: number;
  category: string;
  priority: string;
  status: string;
}
